singer_sdk/connectors/sql.py

```diff
@@ -584,8 +586,10 @@ def schema_exists(self, schema_name: str) -> bool:
         Returns:
             True if the database schema exists, False if not.
         """
-        schema_names = sqlalchemy.inspect(self._engine).get_schema_names()
-        return schema_name in schema_names
+        if schema_name not in self._schema_cache:
+            self._schema_cache = sqlalchemy.inspect(self._engine).get_schema_names()
```

Comment:
Nit: wdyt of making this a set?

```suggestion
            self._schema_cache = set(sqlalchemy.inspect(self._engine).get_schema_names())
```

Reply:
@edgarrmondragon it returned a list of strings https://docs.sqlalchemy.org/en/20/core/reflection.html#sqlalchemy.engine.reflection.Inspector.get_schema_names so I just kept that structure but using a set is probably faster for the lookup, good suggestion!